lib/wda-mac.js

```diff
@@ -334,7 +407,7 @@ class WDAMacServer {
   }
 
   async stopSession () {
-    if (!this.isRunning) {
+    if (!this.isProxyingToRemoteServer && !(this.process?.isRunning)) {
```

Comment:
I assume the condition should be changed to `||`

Reply:
This condition should allow only no `webDriverAgentMacUrl` and the process is not running.
If webDriverAgentMacUrl existed => should always try a deletion request
If the process is running => should try a deletion request